Error while checking build/O2/fullCI for 58240aee363165e199e02598d601040ab5d4eb41 at 2023-02-24 00:16:
```
## sw/BUILD/O2-latest/log
/sw/SOURCES/O2/10828-slc8_x86-64/0/Detectors/TRD/qc/src/Tracking.cxx:137:45: error: 'const class o2::trd::Tracklet64' has no member named 'getSnp'; did you mean 'getSlope'?
/sw/SOURCES/O2/10828-slc8_x86-64/0/Detectors/TRD/qc/src/Tracking.cxx:138:66: error: 'const class o2::trd::Tracklet64' has no member named 'getTgl'
/sw/SOURCES/O2/10828-slc8_x86-64/0/Detectors/TRD/qc/src/Tracking.cxx:138:86: error: 'const class o2::trd::Tracklet64' has no member named 'getTgl'
/sw/SOURCES/O2/10828-slc8_x86-64/0/Detectors/TRD/qc/src/Tracking.cxx:150:48: error: 'const class o2::trd::Tracklet64' has no member named 'getSnp'; did you mean 'getSlope'?
/sw/SOURCES/O2/10828-slc8_x86-64/0/Detectors/TRD/qc/src/Tracking.cxx:151:69: error: 'const class o2::trd::Tracklet64' has no member named 'getTgl'
/sw/SOURCES/O2/10828-slc8_x86-64/0/Detectors/TRD/qc/src/Tracking.cxx:151:86: error: 'const class o2::trd::Tracklet64' has no member named 'getTgl'
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/10828/58240aee363165e199e02598d601040ab5d4eb41/build_O2_fullCI/pretty.html).

--- a/Detectors/TRD/qc/src/Tracking.cxx
+++ b/Detectors/TRD/qc/src/Tracking.cxx
@@ -133,8 +133,13 @@ void Tracking::checkTrack(const TrackTRD& trkTrd, bool isTPCTRD)
       tracklet.getQ2(),
     };
 
-    // Corrected Tracklets
-    auto cor = mLocalGain.getValue(tracklet.getHCID() / 2, tracklet.getPadCol(), tracklet.getPadRow()) * (std::abs(tracklet.getSlopeBinSigned()) + 1.f);
+    /// Corrected Tracklets
+    // To correct for longer drift lengths in the drfit chamber and hence more energy deposition,
+    // we calculate the variation to reference for the tracklets (exrapolated from the track fit).
+    // \sqrt{ (dx/dx)^2 + (dy/dx)^2 + (dz/dx)^2}
+    auto tphi = trk.getSnp() / std::sqrt((1.f - trk.getSnp()) * (1.f + trk.getSnp()));
+    auto trackletLength = std::sqrt(1.f + tphi * tphi + trk.getTgl() * trk.getTgl());
+    auto cor = mLocalGain.getValue(tracklet.getHCID() / 2, tracklet.getPadCol(), tracklet.getPadRow()) * trackletLength;
     float q0{tracklet.getQ0() / cor}, q1{tracklet.getQ1() / cor}, q2{tracklet.getQ2() / cor};
 
     // z-row merging
@@ -145,7 +150,7 @@ void Tracking::checkTrack(const TrackTRD& trkTrd, bool isTPCTRD)
         }
         if (std::abs(tracklet.getPadCol() - trklt.getPadCol()) == 1 && std::abs(tracklet.getPadRow() - trklt.getPadRow()) == 0) {
           // Add charge information
-          auto cor = mLocalGain.getValue(trklt.getHCID() / 2, trklt.getPadCol(), trklt.getPadRow()) * (std::abs(trklt.getSlopeBinSigned()) + 1.f);
+          auto cor = mLocalGain.getValue(trklt.getHCID() / 2, trklt.getPadCol(), trklt.getPadRow()) * trackletLength;
           q0 += trklt.getQ0() / cor;
           q1 += trklt.getQ1() / cor;
           q2 += trklt.getQ2() / cor;

